distributed/tests/py3_test_client.py

```diff
@@ -37,3 +38,30 @@ def test_as_completed_async_for(c, s, a, b):
     yield f()
 
     assert set(results) == set(range(1, 11))
+
+
+def test_asyncio_await():
+    from tornado.platform.asyncio import to_asyncio_future
+    ioloop = asyncio.get_event_loop()
+
+    async def f():
+        scheduler = Scheduler(loop=ioloop)
+        scheduler.start('127.0.0.1')
+        worker = Worker(scheduler.address, loop=ioloop)
+        await to_asyncio_future(worker._start())
+
+        client = Client(scheduler.address, loop=ioloop, start=False)
+        await to_asyncio_future(client._start())
+
+        future = client.submit(inc, 1)
+        result = await future
+        assert result == 2
+
+        await to_asyncio_future(client._shutdown())
```

Comment:
https://github.com/dask/distributed/blob/master/distributed/client.py#L659

What do You think about decorating gen.coroutine methods with an additional to_asyncio_future based on the same ioloop typecheck?